llvm/lib/Target/AMDGPU/VOP2Instructions.td

```diff
@@ -467,6 +479,42 @@ class VOP_MAC <ValueType vt0, ValueType vt1=vt0> : VOPProfile <[vt0, vt1, vt1, v
 }
 
 def VOP_MAC_F16 : VOP_MAC <f16>;
+def VOP_MAC_F16_t16 : VOP_MAC <f16> {
```

Comment:
This is a lot of copy paste. Is there a way to avoid replicating so many lets?

Reply:
I don't see a way to reduce it much. These lets are pretty much standard for a true16 instruction. But we have the true16 lets, and all the VOP_MAC lets to support. We could inherit from VOPProfile_TRUE16 instead of VOP_MAC, but then we'd have all the same lets here as in VOP_MAC.  One upside is when the default is switched to true16, we can delete 6 lines related to VOP3DPP.

Reply:
Thanks Joe for the explanation. let me double check if there are some lines can be removed, but for the majority of them I think they are inevitable